### DISCUSSION TITLE
Why not support extending RouteRecordRaw?

### DISCUSSION THREAD

**Discussion by Hades-Lei** [vuejs/router#1896] [Ideas]
### What problem is this solving

Why not support extending RouteRecordRaw?

### Proposed solution

Why not support extending RouteRecordRaw?

**Comment by posva** [comment]
You can extend route records with `meta`. What other things do you want to add? If you have something more concrete in mind, you can follow the RFC format from https://github.com/vuejs/rfcs/ and open a discussion here

**Comment by biluokong** [comment]
I also encountered this problem.
![image](https://github.com/vuejs/router/assets/95461610/5ea3a57b-1c59-4624-adf1-792141643fb1)

![image](https://github.com/vuejs/router/assets/95461610/1b67779d-4b72-4f0e-aabb-c922c64dbeff)

**Reply by posva** [reply]
You probably have to import it as a type. Just put type in front of the import name

**Reply by biluokong** [reply]
Thank you, it has been resolved
